src/obd/obd.cpp

```diff
@@ -136,18 +170,25 @@ std::string OBD::_read()
 {
     char buf[1];
     std::string str;
-
+  
     while (true) {
-        if (read(this->fd, (void *)buf, 1) != 1) {
-            std::cout << "failed read" << std::endl;
-            this->connected = false;
-            return "";
+        if (((adapterType==BT)?(::recv(btSocket->socketDescriptor(), (void *) buf, 1, 0)):(read(this->fd, (void *)buf, 1))) == -1) {
+            // std::cout << "failed read" << std::endl;
+            if(adapterType != BT){
+                this->connected = false;
+                return "";
+            }else{
+                usleep(500000); //bt is a little slower than USB in this case it seems, give it a little rest
```

Comment:
is there a particular reason youre sleeping here? When I tested with a Bluetooth adapter, it was definitely slower than USB but it wasnt necessary to sleep after every byte read.

Even if you're waiting for data, shouldnt you just be able to set a timeout?

Reply:
Oh yea that uhhhh, should be able to be removed - just waiting for data, but forgot I had added the adapter if/else - was originally added as a retry before setting connected to false.